### DISCUSSION TITLE
HTML File_New Learner_Coding Error Query

### DISCUSSION THREAD

**Discussion by Froggy1977** [community/community#146852] [Programming Help] [answered]
### Body

Hi, I am new to creating websites and have started a basic course. I have a question about the following HTML exercise. Hope you are kind enough to spot my mistake. I am building a website using the bootstrap grid and need to create a two column food menu. I have Set up the Bootstrap container, and should display the logo in the top center of the webpage using Bootstrap and display the food menu in two columns using Bootstrap Grid. However, from the code I have created and when comparing it to the 'answer' (which is the same) the logo and menu are showing down the left side of the page. Is this an issue with Bootstrap (which I doubt)? Can you find my error? Grateful for any responses. 

<body>
    <div class="container">
        <div class="row">
            <div class="col-12">
                <div class="text-center">
                    <img src="logo.png" class="img-fluid">
                </div>
            </div>
        </div>
        <div class="row">
            <div class="col-12">
                <div class="text-center">
                    <h1>Our Menu</h1>
                </div>
            </div>
        </div>
        <div class="row">
            <div class="col-12 col-lg-6">
                <div class="menu-item">
                    <h2>Falafel</h2>
                    <p>Chickpea, herbs, spices.</p>
                </div>
                <div class="menu-item">
                    <h2>Fried Calamari</h2>
                    <p>Squid, buttermilk.</p>
                </div>
            </div>
            <div class="col-12 col-lg-6">
                <div class="menu-item">
                    <h2>Pasta Salad</h2>
                    <p>Lettuce, vegetables, mozzarella.</p>
                </div>
                <div class="menu-item">
                    <h2>Greek Salad</h2>
                    <p>Cucumbers, onion, feta cheese.</p>
                </div>
            </div>
        </div>
  
![Github_correct positioning](https://github.com/user-attachments/assets/0888d036-5881-4019-9930-6583116099eb)
![Github_Incorrect Positioning](https://github.com/user-attachments/assets/fddd7e53-7c8e-4e91-afb6-096bab36a093)
</body>

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by harmeetsingh11** [comment, marked as answer]
Hi @Froggy1977,

The issue described, where the logo and menu content are aligned to the left side of the page rather than in the center or in two columns, is likely related to incorrect usage of Bootstrap classes or missing parent container and row/column structure. Here’s how to debug and fix this problem:

### Diagnosis
1. **Missing Bootstrap Container Class**: Ensure you’ve wrapped the content in a Bootstrap `.container` or `.container-fluid` class. Without this, Bootstrap's grid system will not function properly.
2. **Incorrect Use of Grid Classes**: For two columns, each column must be wrapped in a `.row` and assigned appropriate grid classes like `.col-md-6` or `.col-6`.
3. **Alignment Issue**: For centering the logo, you might need `.text-center` on the logo container.

### Correct Bootstrap Implementation

```html
<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Bootstrap Grid Example</title>
  <link href=" Paste bootstrap min.css link" rel="stylesheet">
</head>
<body>
  <div class="container text-center">
    <!-- Logo Section -->
    <div class="row">
      <div class="col-12">
        <img src="logo.png" alt="Little Lemon Logo" class="img-fluid" style="max-width: 150px;">
      </div>
    </div>

    <!-- Title -->
    <div class="row">
      <div class="col-12">
        <h2 class="my-4">Our Menu</h2>
      </div>
    </div>

    <!-- Menu Section -->
    <div class="row">
      <!-- Left Column -->
      <div class="col-md-6 text-start">
        <h4>Falafel</h4>
        <p>Chickpea, herbs, spices.</p>
        <h4>Fried Calamari</h4>
        <p>Squid, buttermilk.</p>
      </div>
      <!-- Right Column -->
      <div class="col-md-6 text-start">
        <h4>Pasta Salad</h4>
        <p>Lettuce, vegetables, mozzarella.</p>
        <h4>Greek Salad</h4>
        <p>Cucumbers, onion, feta cheese.</p>
      </div>
    </div>
  </div>

  <script src=" Paste bootstrap min.js link"></script>
</body>
</html>
```
### Key Adjustments
1. **Logo Centering**: The `.text-center` class ensures the logo is centered.
2. **Two-Column Layout**: The `.row` and `.col-md-6` classes split the menu into two equal columns on medium and larger screens.
3. **Spacing**: Use margin utilities like `my-4` for spacing between sections.

### Common Pitfalls to Avoid in future
- **Not Including Bootstrap CDN**: Ensure the Bootstrap CSS and JavaScript files are correctly linked.
- **Improper Nesting**: Avoid placing content directly inside a `.container` without a `.row` and appropriate `.col-*` classes.

If this structure still doesn't work, double-check the browser's developer tools (`Ctrl+Shift+I`) to inspect the applied classes and ensure Bootstrap is properly loaded. Let me know if the error is resolved.

Thanks

**Reply by Froggy1977** [reply]
Thanks. Your solution worked.
